### ISSUE TITLE
Typescript should give warnings on types that can be type coerced.

### ISSUE THREAD

**Issue by akgupta0777** [microsoft/TypeScript#44957]
# Bug Report

### 🔎 Search Terms

Typescript type coercion, typescript warning
### 🕗 Version & Regression Information

Typescript version -> 4.2.4

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?#code/PTAEFkE9QYwewCYFNQHM5IM6gBZIE5ICwAUAGYCuAdjAC4CWcVscAtgA4CGhAKgO5wAapwA2FLAAoqFVgEYAXJlr56VVAB9prAEYEANFoBM80Ft34AlAG9SoO6HpkpM2aAA8AXlMzD12-YD4Kkw4ESQAOhE4VGc5UABqUAAiB2wwzEwCUFocTmYk+KMLAG5-OwBfUkqSUjDaUAA3UXEFJRU1UC8kgFYk0pI6xuakYzMsr1kABn74Dm4kfiFhzAkmsSRZPTXxX2KgA)

### 💻 Code

```ts
function compareTwoValues(num1:string|number,num2: number){
    if(num1 <= num2){
        console.log(num1 + " is lesser than "+num2);
    }
}

let value1:string = "5";
let value2:number = 10;
compareTwoValues(value1,value2);
```

### 🙁 Actual behavior

In the function compareTwoValues() **num1** can be either be **string** or a **number** and it is getting compared to a **num2** which is a **number**. 
When comparing num1 with num2, num1 gets type coerced to number and then it do the comparison.
Typescript compiles with no errors and warnings.

### 🙂 Expected behavior

Typescript should give me a warning

**Comment by RyanCavanaugh** [comment]
Duplicate #41163

**Comment by akgupta0777** [comment]
@RyanCavanaugh  Sorry I searched a lot about this issue but didn't found the duplicate. Should I delete this issue ?

**Comment by fatcerberus** [comment]
He linked the issue this is a duplicate of in his message.
